Sources/Instrumentation/Sessions/SessionManager.swift

```diff
@@ -38,22 +30,38 @@ public class SessionManager {
   /// - Returns: The current active session
   @discardableResult
   public func getSession() -> Session {
-    refreshSession()
+    let transition : SessionTransition? = lock.withLock {
+      if session == nil || session!.isExpired() {
+        return startSession()
```

Comment:
calling out here can be really dangerous since you hold the lock. Sometimes you can do it when you have a naming convention like `locked_startSession` so everyone knows that you're already lock in there. The biggest fear is that `startSession` calls out in some fashion. If this isn't the case then ignore.